### ISSUE TITLE
Interfaces and Abstract classes

### ISSUE THREAD

**Issue by vasily-kartashov** [vimeo/psalm#1065]
The following code

```
<?php

interface ExampleInterface
{
    public function f(): array;
}

abstract class ExampleAbstractClass implements ExampleInterface
{
    public function g(): array
    {
        return $this->f();
    }
}
```

will produce following inspection results

```
ERROR: MixedInferredReturnType - Example.php:10:26 - Could not verify return type 'array<mixed, mixed>' for ExampleAbstractClass::g
    public function g(): array
    {
        return $this->f();
    }


ERROR: MixedReturnStatement - Example.php:12:9 - Could not infer a return type
        return $this->f();
```

Both seem to be false positives.

**Comment by muglug** [comment]
Reproduced here: https://getpsalm.org/r/890597949c
